### PR TITLE
[Checkstyle] Use prepare-commit-msg hook for checkstyle

### PR DESCRIPTION
**Pull request by avano**
//skip-ci

### CODE REVIEW

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by avano** [comment]
@asmigala do you agree with the configuration?

File: checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        <!-- See http://checkstyle.sourceforge.net/config_misc.html#UpperEll -->
+        <module name="UpperEll"/>
+        <!-- Checks that a token is followed by whitespace -->
+        <!-- See http://checkstyle.sourceforge.net/config_whitespace.html#WhitespaceAfter -->
+        <module name="WhitespaceAfter"/>
+        <!-- Checks that a token is surrounded by whitespace -->
+        <!-- See http://checkstyle.sourceforge.net/config_whitespace.html#WhitespaceAround -->
+        <module name="WhitespaceAround"/>
+    </module>
+    <!-- See http://checkstyle.sourceforge.net/config_imports.html#FileTabCharacter -->
```
Comment (asmigala) [inline checkstyle/checkstyle.xml]: This is not the right URL, should be `http://checkstyle.sourceforge.net/config_whitespace.html#FileTabCharacter`

File: hooks/prepare-commit-msg [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+CHECKSTYLE="checkstyle-8.20"
+CHECKSTYLE_JAR="${CHECKSTYLE}-all.jar"
+ROOT_DIR=$(git rev-parse --show-toplevel)
+
+FILES=$(git diff --cached --name-only --diff-filter=ACM | cat)
+
+COMMIT_MSG="$(cat "$1")"
+
+# Skip check if the commit msg starts with "tmp"
```
Comment (asmigala) [inline hooks/prepare-commit-msg]: maybe put this in the readme? not a big deal though

Comment (avano) [reply hooks/prepare-commit-msg]: yup, I'll add a paragraph in the readme

**Comment by avano** [comment]
also I have the intellij settings exported that conform the checkstyle configuration.. currently it is hosted in our internal gitlab, but probably it doesn't make sense to have it there and it would be better to add it to the `checkstyle` folder in syndesis-qe, wdyt @asmigala?

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by asmigala** [comment]
Agree on adding the intellij settings here. 

I agree on all the checks that are enabled, but I'd suggest adding some more: 

http://checkstyle.sourceforge.net/config_coding.html#HiddenField
http://checkstyle.sourceforge.net/config_coding.html#MagicNumber
http://checkstyle.sourceforge.net/config_coding.html#SimplifyBooleanExpression
http://checkstyle.sourceforge.net/config_coding.html#SimplifyBooleanReturn
http://checkstyle.sourceforge.net/config_coding.html#UnnecessaryParentheses

(some of these might be too drastic for an existing code base, e.g. the MagicNumber one)

**Comment by avano** [comment]
we had magic number in our old testsuite and what I ended up with was `final long second = 1000L` :grinning: 
we can add it as a warning only, but as always, everyone will ignore warnings

I'll add the other 4 and will skip the magic number, agreed?

**Comment by asmigala** [comment]
:D you're right, that would be annoying. There's a `ignoreNumbers` property on the check, but you'd still have to `wait(FIVE * SECONDS)`.

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by asmigala** [comment]
lgtm, will approve once it's not a draft

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by avano** [comment]
@asmigala ready

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Pull request merged**